core/core.go

```diff
@@ -296,7 +297,22 @@ func getBazelVersion() (string, error) {
 		}
 	}
 
-	return "latest", nil
+	bazelFallbackVersion := GetEnvOrConfig("USE_BAZEL_FALLBACK_VERSION")
+	if len(bazelFallbackVersion) == 0 {
+		bazelFallbackVersion = "silent:latest"
+	}
+	if strings.HasPrefix(bazelFallbackVersion, "error:") {
+		bazelFallbackVersion = bazelFallbackVersion[6:]
+		return "", fmt.Errorf("not allowed to use fallback version %q", bazelFallbackVersion)
+	} else if strings.HasPrefix(bazelFallbackVersion, "warn:") {
+		bazelFallbackVersion = bazelFallbackVersion[5:]
+		log.Printf("Warning: used fallback version %q\n", bazelFallbackVersion)
+	} else if strings.HasPrefix(bazelFallbackVersion, "silent:") {
+		bazelFallbackVersion = bazelFallbackVersion[7:]
+	} else {
+		return "", fmt.Errorf("invalid fallback version format %q", bazelFallbackVersion)
+	}
+	return bazelFallbackVersion, nil
```

Comment:
nit: generally in Golang we prefer to avoid using `else / else if` where possible.

Here you could rewrite each of the `else / else if` clause with an early `return` which should make the code a lot easier to read.

Also the reassignment of `bazelFallbackVersion` could generally be avoided to improve readability as well.

Here is how I would rewrite this block to make it easier to read.
```
	bazelFallbackVersion := GetEnvOrConfig("USE_BAZEL_FALLBACK_VERSION")
	if len(bazelFallbackVersion) != 0 {
		if strings.HasPrefix(bazelFallbackVersion, "error:") {
			return "", fmt.Errorf("not allowed to use fallback version %q", bazelFallbackVersion[6:])
		}
		if strings.HasPrefix(bazelFallbackVersion, "warn:") {
			v := bazelFallbackVersion[5:]
			log.Printf("Warning: used fallback version %q\n", v)

			return v, nil
		}
		if strings.HasPrefix(bazelFallbackVersion, "silent:") {
			return bazelFallbackVersion[7:], nil
		}

		return "", fmt.Errorf("invalid fallback version format %q", bazelFallbackVersion)
	}

	return "latest", nil
```

Reply:
can probably do `s/bazelFallbackVersion/fallbackVersion/` as well